src/odemis/acq/path.py

```diff
@@ -558,6 +577,19 @@ def _doSetPath(self, path, detector):
                     continue
                 if not hasattr(comp, "axes") or not isinstance(comp.axes, dict):
                     continue
+                if isinstance(pos, tuple):  # several ways to find the position
+                    for position in pos:
+                        if isinstance(position, str) and position.startswith("MD:"):
+                            try:
+                                pos = self.mdToValue(comp, position[3:])[axis]
+                                break
+                            except KeyError:
+                                pass
+                        else:
+                            pos = position
+                            break
```

Comment:
Would this part not already simplify if we write like this?
With he whole for loop being some many if statements and try excepts it becomes commit by commit harder to read.We could create sub methods for each part or something else. But this might already help a bit.

 

>                        if isinstance(position, dict) or  isinstance(position, float):
>                             pos = position
>                             break
>                        else:
>                              try:
>                                   pos = self.mdToValue(comp, position[3:])[axis]
>                                   break
>                               except KeyError:
>                                   pass